titus-server-master/src/main/java/com/netflix/titus/master/jobmanager/service/event/JobEventFactory.java

```diff
@@ -82,7 +90,24 @@ public JobManagerReconcilerEvent newModelUpdateErrorEvent(ReconciliationEngine<J
                                                               EntityHolder previousEntityHolder,
                                                               Throwable error,
                                                               String transactionId) {
-        return new JobModelReconcilerEvent.JobModelUpdateErrorReconcilerEvent(getJob(engine, modelActionHolder), (TitusChangeAction) changeAction, modelActionHolder, previousEntityHolder, error, transactionId);
+        return new JobModelReconcilerEvent.JobModelUpdateErrorReconcilerEvent(getJob(engine, modelActionHolder), (TitusChangeAction) changeAction, modelActionHolder, previousEntityHolder, error, transactionId,
+                getCallMetadata(engine, modelActionHolder));
+    }
+
+    private CallMetadata getCallMetadata(ReconciliationEngine<JobManagerReconcilerEvent> engine, ModelActionHolder modelActionHolder) {
+        switch (modelActionHolder.getModel()) {
+            case Running:
+                if (engine.getRunningView().getAttributes().get(JobAttributes.JOB_ATTRIBUTE_CALLMETADATA) == null){
+                    return UNDEFINED_CALL_METADATA;
+                }
+            case Store:
```

Comment:
Why do we check `case Store` and `engine.getStoreView()` when we only ever return `engine.getRunningView()`?